alonzo/impl/src/Cardano/Ledger/Alonzo/Tx.hs

```diff
@@ -412,8 +315,27 @@ isTwoPhaseScriptAddress tx addr =
         Just scr -> not (isNativeScript @era scr)
 
 -- | txsize computes the length of the serialised bytes
-instance HasField "txsize" (ValidatedTx era) Integer where
-  getField (ValidatedTxConstr (Memo _ bytes)) = fromIntegral (SBS.length bytes)
+instance
+  ( Typeable era,
+    ToCBOR (Core.TxBody era),
+    ToCBOR (Core.AuxiliaryData era)
+  ) =>
+  HasField "txsize" (ValidatedTx era) Integer
+  where
+  getField tx = fromIntegral (BS.length $ serialize' tx)
+
+instance
```

Comment:
I think the change here is good, but sufficiently subtle that it needs some documentation. There are three things we should point out:

1. This instance only exists for the purposes of size calculation. A `ValidatedTx` is never serialised/deserialised naked (hence the lack of a corresponding `FromCBOR` instance), only as part of a `TxSeq`.
2. It is legitimate to do this for the purposes of size calculation only because the individual components all actually store their bytes; thus, the only work we do in this function is concatenating them.
3. The reason we only encode three parts of the transaction (throwing away the `IsValidating` tag) is for easy compatibility with Mary, ensuring that upgraded transactions stay the same size. The cost of this is very small and amortised over multiple transactions anyway, so it's not a problem to ignore it.

Reply:
Jared notices a fourth one:
By lucky coincidence, this size is the same as the size of the tx submitted to the mempool.